src/infrasys/function_data.py

```diff
@@ -0,0 +1,113 @@
+from infrasys import Component
+from typing_extensions import Annotated
+from pydantic import Field, model_validator
+from pydantic.functional_validators import AfterValidator
+from typing import NamedTuple
+import numpy as np
+
+
+class XY_COORDS(NamedTuple):
+    x: float
+    y: float
+
+
+class LinearFunctionData(Component):
+    """
+    Class to represent the underlying data of linear functions. Principally used for
+    the representation of cost functions `f(x) = proportional_term*x + constant_term`.
+    """
+
+    name: Annotated[str, Field(frozen=True)] = ""
+    # match above annotation
+    proportional_term: Annotated[
+        float, Field(description="the proportional term in the represented function")
+    ]
+    constant_term: Annotated[
+        float, Field(description="the constant term in the represented function")
+    ]
+
+
+class QuadraticFunctionData(Component):
+    """
+    Class to represent the underlying data of quadratic functions. Principally used for the
+    representation of cost functions
+    `f(x) = quadratic_term*x^2 + proportional_term*x + constant_term`.
+    """
+
+    name: Annotated[str, Field(frozen=True)] = ""
+    quadratic_term: Annotated[
+        float, Field(description="the quadratic term in the represented function")
+    ]
+    proportional_term: Annotated[
+        float, Field(description="the proportional term in the represented function")
+    ]
+    constant_term: Annotated[
+        float, Field(description="the constant term in the represented function")
+    ]
+
+
+def validate_piecewise_x(points: list):
```

Comment:
Could you add docstring to the function? See this example:  https://numpydoc.readthedocs.io/en/latest/example.html